### PR TITLE
SIMSBIOHUB-544: Update project/survey start/end date constraints

### PR DESCRIPTION
**Pull request by NickPhura**
Update associated yup schema.

## Links to Jira Tickets

https://apps.nrs.gov.bc.ca/int/jira/browse/SIMSBIOHUB-544

## Description of Changes

Update project and survey date triggers to remove restrictions on survey dates being between project dates.
Update associated yup schema.

Why? The triggers were not complete (there were cases they didn't handle correctly) + the project dates are going to be deprecated in future. So this resolves the trigger issue (by removing the constraints altogether), and moves a few steps closer to the eventual goal of removing the project dates.

## Testing Notes

Project start date must always come before its own end date, as before.

Survey start date must always come before its own end date, as before, but now has no constraints related to the project dates. So a survey could start before a project, or start after the project has ended, etc.

### CODE REVIEW

File: database/src/procedures/delete_survey_procedure.ts
Comment (NickPhura) [inline database/src/procedures/delete_survey_procedure.ts:1]: I just ran the vscode 'Trim Trailing Whitespace' command to satisfy the `--sql` template extension.

File: database/src/procedures/delete_project_procedure.ts
Comment (NickPhura) [inline database/src/procedures/delete_project_procedure.ts:1]: I just ran the vscode 'Trim Trailing Whitespace' command to satisfy the `--sql` template extension.

**Review by al-rosenthal — COMMENTED** [review]
the new trigger works but I noticed that the survey end date component is still warning that `End Date cannot come after the Project End Date`

**Review by mauberti-bc — APPROVED** [review]
Created Projects and Surveys with non-overlapping dates 👍

**Review by curtisupshall — APPROVED** [review]
Code looks good to me

**Pull request merged**